src/main/java/org/jruby/ext/openssl/SSLContext.java

```diff
@@ -517,17 +594,47 @@ public IRubyObject set_ssl_version(IRubyObject version) {
         } else {
             versionStr = version.convertToString().toString();
         }
+        set_ssl_version(versionStr);
+        return version;
+    }
+
+    private void set_ssl_version(final String versionStr) {
         final String protocol = SSL_VERSION_OSSL2JSSE.get(versionStr);
         if ( protocol == null ) {
             throw getRuntime().newArgumentError("unknown SSL method `"+ versionStr +"'");
         }
         this.protocol = protocol;
+        this.protocolVersion = METHODS_MAP.get(versionStr);
         protocolForServer = ! versionStr.endsWith("_client");
         protocolForClient = ! versionStr.endsWith("_server");
-        return version;
     }
 
-    final String getProtocol() { return this.protocol; }
+    @JRubyMethod(name = "set_minmax_proto_version")
+    public IRubyObject set_minmax_proto_version(ThreadContext context, IRubyObject minVersion, IRubyObject maxVersion) {
+        minProtocolVersion = parseProtoVersion(minVersion);
+        maxProtocolVersion = parseProtoVersion(maxVersion);
+
+        return context.nil;
+    }
+
+    private int parseProtoVersion(IRubyObject version) {
+        if (version.isNil())
+            return 0;
+        if (version instanceof RubyFixnum) {
+            return (int) ((RubyFixnum) version).getLongValue();
```

Comment:
let's just to `getIntValue()` here so we handle overflown values

Reply:
Unfortunately there is no `getIntValue()`.
But I double checked and found `static int fix2int(RubyFixnum arg)`

Reply:
right, yeah that one doing the `int` check before cast.

Reply:
Done